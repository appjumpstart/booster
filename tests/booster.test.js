const { join } = require('path')

const execa = require('execa')
const Knex = require('knex')
const config = require('./fixtures/knexfile')

const booster = '../../index.js'
const cwd = join(__dirname, 'fixtures')

async function assertDbSeeded (knex) {
  const { rows } = await knex.raw('SELECT * FROM categories')
  expect(rows[0].name).toBe('Personal')
  expect(rows[1].name).toBe('Business')
}

test('booster migrates and seeds the default database', async () => {
  const knex = Knex(config.test)
  try {
    const { stdout } = await execa(booster, { cwd })
    expect(stdout).toMatchSnapshot()
    await assertDbSeeded(knex)
  } finally {
    await knex.destroy()
  }
})

test('booster migrates and seeds the specified database', async () => {
  const knex = Knex(config.development)
  try {
    const { stdout } = await execa(booster, ['development'], { cwd })
    expect(stdout).toMatchSnapshot()
    await assertDbSeeded(knex)
  } finally {
    await knex.destroy()
  }
})

test('booster migrates and seeds using the specified knexfile', async () => {
  const knex = Knex(config.test)
  try {
    const { stdout } = await execa(booster, ['-c', cwd], { cwd })
    expect(stdout).toMatchSnapshot()
    await assertDbSeeded(knex)
  } finally {
    await knex.destroy()
  }
})

test('booster outputs error when specified knexfile not found', async () => {
  const cmd = execa(booster, ['-c', '/tmp'], { cwd })
  await expect(cmd).rejects.toThrowErrorMatchingSnapshot()
})
