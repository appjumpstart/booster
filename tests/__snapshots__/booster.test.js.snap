// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`booster migrates and seeds the default database 1`] = `
"
  ğŸ‘Ÿ Migrations run!

  ğŸŒ± Database seeded!"
`;

exports[`booster migrates and seeds the specified database 1`] = `
"
  ğŸ‘Ÿ Migrations run!

  ğŸŒ± Database seeded!"
`;

exports[`booster migrates and seeds using the specified knexfile 1`] = `
"
  ğŸ‘Ÿ Migrations run!

  ğŸŒ± Database seeded!"
`;

exports[`booster outputs error when specified knexfile not found 1`] = `
"Command failed: ../../index.js -c /tmp

  ğŸš« Could not find knexfile.js at /tmp/knexfile.js

"
`;
